x-pack/libbeat/management/managerV2.go

```diff
@@ -642,15 +659,57 @@ func (cm *BeatV2Manager) reloadInputs(inputUnits []*client.Unit) error {
 		inputBeatCfgs = append(inputBeatCfgs, inputCfg...)
 	}
 
-	if !didChange(cm.lastInputCfgs, inputCfgs) {
+	if !didChange(cm.lastInputCfgs, inputCfgs) && !cm.forceReload.Load() {
 		cm.logger.Debug("Skipped reloading input units; configuration didn't change")
 		return nil
 	}
 
-	err := obj.Reload(inputBeatCfgs)
-	if err != nil {
-		return fmt.Errorf("failed to reloading inputs: %w", err)
+	if cm.forceReload.Load() {
+		cm.logger.Info("Reloading Beats inputs because forceReload is true. " +
+			"Set log level to debug to get more information about which " +
+			"inputs are causing this.")
+	}
+
+	if err := obj.Reload(inputBeatCfgs); err != nil {
+		merror := &multierror.MultiError{}
+		realErrors := multierror.Errors{}
+
+		// At the moment this logic is tightly bound to the current RunnerList
+		// implementation from libbeat/cfgfile/list.go and Input.loadStates from
+		// filebeat/input/log/input.go.
+		// If they change the way they report errors, this will break.
+		// TODO (Tiago): update all layers to use the most recent features from
+		// the standard library errors package.
+		if errors.As(err, &merror) {
+			for _, err := range merror.Errors {
+				causeErr := cause(err)
+				// A Log input is only marked as finished when all events it
+				// produceds are acked by the acker so when we see this error,
+				// we just retry until the new input can be started.
+				// This is the same logic used by the standalone configuration file
+				// reloader implemented on libbeat/cfgfile/reload.go
+				inputNotFinishedErr := &common.ErrInputNotFinished{}
+				if ok := errors.As(causeErr, &inputNotFinishedErr); ok {
+					cm.logger.Debugf("file '%s' is not finished, will retry starting the input soon", inputNotFinishedErr.File)
+					cm.forceReload.Store(true)
+					cm.logger.Debug("ForceReload set to TRUE")
```

Comment:
Do you need this extra log line? Isn't it clear from the log line before the `.Store(true)` that is what had happened.

Reply:
It pretty much is, I decided to be a quite verbose here to help understand why inputs are not starting even though Filebeat is still reporting health. Also I wanted a matching line for 
https://github.com/elastic/beats/blob/0b36d196a4afcce7b0d3bef63eea1b3b7d962b3e/x-pack/libbeat/management/managerV2.go#L710

So the state changes of `cm.forceReload` are very clear in the logs.

Reply:
Okay.